Kernel/Arch/x86/common/Interrupts.cpp

```diff
@@ -317,14 +317,19 @@ void page_fault_handler(TrapFrame* trap)
     PageFault fault { regs.exception_code, VirtualAddress { fault_address } };
     auto response = MM.handle_page_fault(fault);
 
-    if (response == PageFaultResponse::ShouldCrash || response == PageFaultResponse::OutOfMemory) {
+    if (response == PageFaultResponse::ShouldCrash || response == PageFaultResponse::OutOfMemory || response == PageFaultResponse::BusError) {
         if (faulted_in_kernel && handle_safe_access_fault(regs, fault_address)) {
             // If this would be a ring0 (kernel) fault and the fault was triggered by
             // safe_memcpy, safe_strnlen, or safe_memset then we resume execution at
             // the appropriate _fault label rather than crashing
             return;
         }
 
+        if (response == PageFaultResponse::BusError && current_thread->has_signal_handler(SIGBUS)) {
```

Comment:
what if the current thread does not have a signal handler `SIGBUS`? the code below will crash it a SIGSEGV instead of a SIGBUS as the cause, that needs to be updated as well.